src/main/java/com/aws/greengrass/util/RegionUtils.java

```diff
@@ -47,10 +53,15 @@ public static String getGreengrassControlPlaneEndpoint(String awsRegion,
      * @param port endpoint port
      * @return Greengrass ServiceEndpoint
      */
-    public static String getGreengrassDataPlaneEndpoint(String awsRegion,
-                                                        IotSdkClientFactory.EnvironmentStage stage,
+    public static String getGreengrassDataPlaneEndpoint(String awsRegion, IotSdkClientFactory.EnvironmentStage stage,
                                                         int port) {
         String dnsSuffix = Region.of(awsRegion).metadata().partition().dnsSuffix();
+        if (Region.CN_NORTH_1.equals(Region.of(awsRegion))) {
+            // CN_NORTH_1 has a special endpoint format
+            return String
+                    .format(GREENGRASS_DATA_PLANE_STAGE_TO_ENDPOINT_FORMAT_CN_NORTH_1.get(stage), awsRegion, dnsSuffix,
+                            port);
+        }
         return String.format(GREENGRASS_DATA_PLANE_STAGE_TO_ENDPOINT_FORMAT.get(stage), awsRegion, dnsSuffix, port);
```

Comment:
Do we have to hard code the url in release? There are multiple ways we could externalize this.

Reply:
Not quite sure about that, what do you have in mind to externalize this, keeping in mind it shouldn't be cloud?